src/components/shared/ContextMenu.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { ContextMenu as CMenu, MenuItem, ContextMenuTrigger } from 'react-contextmenu';
+
+interface ContextMenuProps {
+  children: React.ReactNode;
+  menuId: string;
+  value: any;
+}
+
+interface MenuItemProps {
+  event: React.TouchEvent<HTMLDivElement> | React.MouseEvent<HTMLDivElement, MouseEvent>;
+  data: Object;
+  target: HTMLElement;
+}
+
+//This doesn't come with default styling unfortunately. They're style sheets are here: https://github.com/vkbansal/react-contextmenu/blob/master/examples/react-contextmenu.css
+export const ContextMenu = ({ children, menuId, value }: ContextMenuProps) => {
+  function handleClick({ event, data, target }: MenuItemProps) {
```

Comment:
I think what you're trying to do is this:
```
interface MenuItemProps {
  event: React.TouchEvent<HTMLDivElement> | React.MouseEvent<HTMLDivElement, MouseEvent>;
  target: HTMLElement;
}

interface FooObject {
  foo: string;
}
...
function handleClick({ target }: MenuItemProps, data: FooObject) {
    console.log(data, target.innerText);
}
```
when you pass in parameters into a function, you can have multiple: 
`function foo(arg1, arg2) {...}`
and then when you destructure them, it looks like this: 
`function foo({arg1prop1, arg1prop2}, {arg2prop1, arg2prop2}) {...}`

also e.target.innerText gets you access to the text inside the html element